packages/frontend/src/components/SwapDetails/SwapDetails.tsx

```diff
@@ -1,6 +1,59 @@
 import { ReactComponent as GasIcon } from 'src/assets/icons/gas.svg';
 import { useGasFee } from 'src/hooks/useGasFee';
 import { motion, AnimatePresence } from 'framer-motion';
+import { useQuote } from 'src/hooks/useQuote';
+import { EitherQuoteSifiAction } from '@sifi/sdk';
+import { ReactComponent as UniswapIcon } from 'src/assets/bridges/uniswap.svg';
+import { ReactComponent as CurveIcon } from 'src/assets/bridges/curve.svg';
+import { ReactComponent as StargateIcon } from 'src/assets/bridges/stargate.svg';
+import { ReactComponent as ArrowRight } from 'src/assets/arrow-right.svg';
+
+type ExchangeDetails = {
+  [key: string]: { icon: JSX.Element; name: string } | undefined;
+};
+
+type Step = {
+  icon: JSX.Element | null;
+  exchangeName: string | null;
+};
+
+const iconClassName = 'w-4 h-4 border-smoke border rounded-full';
+
+const exchangeDetails: ExchangeDetails = {
+  uniswap: { icon: <UniswapIcon className={iconClassName} />, name: 'Uniswap' },
+  stargate: { icon: <StargateIcon className={iconClassName} />, name: 'Stargate' },
+  curve: { icon: <CurveIcon className={iconClassName} />, name: 'Curve' },
+};
+
+const getIconForExchange = (action: EitherQuoteSifiAction) => {
+  if ('exchange' in action) {
+    const exchange = action.exchange.toLowerCase();
+
+    for (let key in exchangeDetails) {
+      const regex = new RegExp(key, 'i');
+
+      if (regex.test(exchange)) {
+        return exchangeDetails[key]?.icon || null;
+      }
+    }
+  }
+  return null;
+};
+
+const getExhangeName = (action: EitherQuoteSifiAction): string | null => {
+  if ('exchange' in action) {
+    const exchange = action.exchange.toLowerCase();
+
+    for (let key in exchangeDetails) {
+      const regex = new RegExp(key, 'i');
+
+      if (regex.test(exchange)) {
+        return exchangeDetails[key]?.name || null;
+      }
+    }
+  }
+  return null;
```

Comment:
We could be returning a default "unknown" icon maybe so even if only part of a step is available we still get to see some of it?